### ISSUE TITLE
WslRegisterDistribution failed with error: 0x80004005

### ISSUE THREAD

**Issue by JitterDev** [microsoft/WSL#5440]
Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x80004005
Error: 0x80004005 Unspecified error

Press any key to continue...

This happens on Ubuntu, Ubuntu 1804, and 2004. I haven't tested any other repos
I have Windows Subsystem for Linux on and I can't find anything online about this error
I downloaded all repos I used from Windows Store
**Sorry if this is stupid, I'm inexperienced with WSL and Linux**

**Comment by therealkenc** [comment]
Possibly #4003.

Please post the output of the following from powershell:
```
PS> [System.Environment]::OSVersion
PS> Get-CimInstance -Namespace root/SecurityCenter2 -ClassName AntivirusProduct
```

**Comment by Erik-Sovereign** [comment]
@JitterDev 
Did you activate the "Virtual Machine Platform" Windows feature?
When I didn't, I recieved the same error.

**Comment by JitterDev** [comment]
@Erik-Sovereign Yes, still get the error
@therealkenc This is what I get when I type that:
```powershell
PS C:\Users\xdc20> [System.Environment]::OSVersion

Platform ServicePack Version      VersionString
-------- ----------- -------      -------------
 Win32NT             10.0.18362.0 Microsoft Windows NT 10.0.18362.0


PS C:\Users\xdc20> Get-CimInstance -Namespace root/SecurityCenter2 -ClassName AntivirusProduct


displayName              : Windows Defender
instanceGuid             : {D68DDC3A-831F-4fae-9E44-DA132C1ACF46}
pathToSignedProductExe   : windowsdefender://
pathToSignedReportingExe : %ProgramFiles%\Windows Defender\MsMpeng.exe
productState             : 397568
timestamp                : Sat, 20 Jun 2020 17:14:46 GMT
PSComputerName           :

```

**Comment by lygstate** [comment]
Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x80004005
Error: 0x80004005 ??????

Press any key to continue...

so is me

**Comment by therealkenc** [comment]
Please collect [WSL logs](https://github.com/microsoft/WSL/blob/master/CONTRIBUTING.md#8-collect-wsl-logs) and post the backlink to the feedback item here. Also please identify any third-party networking software that has been installed on Windows.

**Comment by tmknight** [comment]
https://aka.ms/AAaaqz1

**Comment by JitterDev** [comment]
Magically started working again. I don't know why, how, or when.

Windows is weird, man.

**Comment by tmknight** [comment]
Still an issue for me.  On latest (as of this post) Build 20257

**Comment by maxsharov** [comment]
Also check if there is enough of free space on your system drive.

**Comment by lygstate** [comment]
I am using AMD CPU 3950x and windows are running inside qemu-kvm with nested vm enabled.
But still doesn't working.

**Comment by tmknight** [comment]
@maxsharov, bare/base Linux shouldn't require more than (a very generous) 8GB when installing all the server components.  Ubuntu states 2.5GB minimum for server.  The WSL VHD is set to 256GB (dynamic), though, typical install for Ubuntu is less than 2GB.  Do you have a MS recommended minimum?

**Comment by k3ring** [comment]
> I am using AMD CPU 3950x and windows are running inside qemu-kvm with nested vm enabled.
> But still doesn't working.

This error is usually caused by not having enough disk space. In my experience you need quite a bit more than the recommended minimum

**Comment by keitarobr** [comment]
I'm having the same error here. 

`Installing, this may take a few minutes...`
`WslRegisterDistribution failed with error: 0x80004005`

No events registered in Windows Logs and Virtualization is possible (if I enable and run hyper-V, I'm able to create a VM with Ubuntu 20.04 and run it).

Microsoft Windows 10.0.21313.1000

**Comment by keitarobr** [comment]
> I'm having the same error here.
> 
> `Installing, this may take a few minutes...`
> `WslRegisterDistribution failed with error: 0x80004005`
> 
> No events registered in Windows Logs and Virtualization is possible (if I enable and run hyper-V, I'm able to create a VM with Ubuntu 20.04 and run it).
> 
> Microsoft Windows 10.0.21313.1000

I "fixed" my system creating a .wslconfig and disabling nested virtualization:

`[wsl2]`
`nestedVirtualization=false`

**Comment by tmknight** [comment]
AMD processor; the following resolved (https://github.com/microsoft/WSL/issues/5440#issuecomment-778660156)

%userprofile%\.wslconfig
`[wsl2]
nestedVirtualization=false`

**Comment by lygstate** [comment]
> AMD processor; the following resolved ([#5440 (comment)](https://github.com/microsoft/WSL/issues/5440#issuecomment-778660156))
> 
> %userprofile%.wslconfig
> 
> ```
> [wsl2]
> nestedVirtualization=false
> ```

well。we want nest virtualization

**Comment by tmknight** [comment]
Agreed; not saying.  However, until that is fixed, you will have to use that config to use WSL2 on AMD processors.

**Comment by BiniamEyakem** [comment]
I got same error 0x80004005, restart the same process but with Administrative privilege and installation run successfully.

**Comment by jbciardi** [comment]
This happens to me too where it wont install the distro.
But it will only happen when I try to install a distro under WSL2.
When default version is set to WSL1, it will install fine. 
I don't know what's happening to cause this nonsense.

**Comment by d-v** [comment]
Hello getting error 0x80004005 when installing debian on win10 pro. 85gb available on disk, both virtual machine platform and windows subsystem for linux are enabled. Seems to be crashing at onecore\vm\wsl\lxss\lxcmdlineshared\hcs.cpp:244
![image](https://user-images.githubusercontent.com/7275135/147164998-40e4698c-df7a-40ea-a4e2-be94d8d5b99e.png)
Tried running with both user and admin privileges

**Comment by kotenok2000** [comment]
[wsl.etl.zip](https://github.com/microsoft/WSL/files/7799251/wsl.etl.zip)
I have uploaded etl logs using wsl.wprp

**Comment by kotenok2000** [comment]
I have decided to reinstall wsl component and when i deselected it in windows components and restarted I was not able to reinstall it.
here is my cbs.log
[CBS.log.zip](https://github.com/microsoft/WSL/files/7891530/CBS.log.zip)

**Comment by rifaterdemsahin** [comment]
I am there mate > testing your kali linux now

**Comment by rifaterdemsahin** [comment]
Does not work on my end
![image](https://user-images.githubusercontent.com/5270036/174992108-d13f2dc4-6aca-4724-aca6-2d850fbfd193.png)

**Comment by rifaterdemsahin** [comment]
also tried deleting this folder no help
![image](https://user-images.githubusercontent.com/5270036/174993225-4a630b0e-5416-4acb-8902-8f7c7f5b27a2.png)

**Comment by dragonkali** [comment]
Are using app or vm version my kali using app not vm usb live


Pada tanggal Rab, 22 Jun 2022 17:20, Erdem ***@***.***>
menulis:

**Comment by rifaterdemsahin** [comment]
> Are using app or vm version my kali using app not vm usb live

I installed windows server 2022 on a vmware workstation using 3995x amd chip

And i tried all in the swl list none of them worked
![image](https://user-images.githubusercontent.com/5270036/174994166-4a24d4e6-99cf-455e-8707-7efb0f421fc6.png)

![image](https://user-images.githubusercontent.com/5270036/174994198-7e4bb89a-00d3-4211-94d0-c21a318992f6.png)

**Comment by rifaterdemsahin** [comment]
**when i moved to 1** it worked...I need the 2 to run the docker on this box
![image](https://user-images.githubusercontent.com/5270036/174996245-d63039e0-53a1-467f-a42d-f1acbe1fbfea.png)

**Comment by rifaterdemsahin** [comment]
> try using run as administrator

i always run as admin

**Comment by dragonkali** [comment]
but is work now

**Comment by rifaterdemsahin** [comment]
yes it works when i set it to 1 but i need to run windows containers on w2022 so i hope it docker works
![image](https://user-images.githubusercontent.com/5270036/174996663-8bfeaf03-27f6-41da-83db-9cd17b39bff9.png)

**Comment by rifaterdemsahin** [comment]
When i move to 2 than things stop working
![image](https://user-images.githubusercontent.com/5270036/174998594-6950890b-740e-4ea9-aea1-c649a9e6aecc.png)

**Comment by dragonkali** [comment]
maybe this report from member github docker usseful for you https://github.com/docker/for-win/issues/8204

**Comment by dragonkali** [comment]
maybe the problem is your wsl

**Comment by dragonkali** [comment]
ubuntu work for wsl 2

**Comment by dragonkali** [comment]
but i have problem with kali linux gui in wsl 1 there much problem in wsl 1  1. you never in desktop gui kali because xrdp scree is black and stuck in loading screen but ihope in wsl 2 desktop kali gui not stuck in screen and black

**Comment by rifaterdemsahin** [comment]
wsl 2 is the way to go to get the high end support.

I made it work in an azure vm not on my local vmware workstation

**Comment by dragonkali** [comment]
can you explain what it means

**Comment by rifaterdemsahin** [comment]
> wsl 2 is the way to go to get the high end support.
> 
> I made it work in an azure vm not on my local vmware workstation

I am also on 3995x
I think we need to update the ubuntu for the amd64 libraries as they are integrated
![image](https://user-images.githubusercontent.com/5270036/175017291-17a31577-4d92-44e2-b7b1-54331fcb5df2.png)
https://twitter.com/rifaterdemsahin/status/1539569229539803138

**Comment by dragonkali** [comment]
maybe

**Comment by rifaterdemsahin** [comment]
> maybe

I cant run docker it could be related to the amd 
![image](https://user-images.githubusercontent.com/5270036/175022804-7ec426a8-3f6e-4855-845f-21ff02053423.png)

they say i can run it but it does not work

**Comment by rifaterdemsahin** [comment]
> ollowing resolved ([#5440 (c](https://github.com/microsoft/WSL/issues/5440#issuecomment-778660156)

does that work in amd?
i thought we get blocked by this
https://docs.microsoft.com/en-us/system-center/vmm/vm-nested-virtualization?view=sc-vmm-2022

**Comment by rifaterdemsahin** [comment]
> nestedVirtualization=false

![image](https://user-images.githubusercontent.com/5270036/175024207-f542ea3a-8092-4bef-b120-45c1746ab4ac.png)


so what happens when i do this in vmware workstation ?

**Comment by rifaterdemsahin** [comment]
> `[wsl2]`
> `nestedVirtualization=false`

how come this worked ?

**Comment by camomilesson** [comment]
> Also check if there is enough of free space on your system drive.

Just encountered this problem and nothing worked. Turned out this user was right, it's the disc space! Fixed

**Comment by djehutyy** [comment]
` wsl --update` as administrator worked for me

**Comment by LukasMeine** [comment]
@djehutyy This actually worked! Been struggling with this for hours, thanks man!

**Comment by renanlJ** [comment]
> ` wsl --update` as administrator worked for me
> 
> Then just `wsl --install -d "the distro you want to install"`

It´s work for me, thanks bro!

**Comment by NuRS7** [comment]
I dont know... I do not know Kali does not want to work on the first and on the second. Plz help me. TG @t.me/NuRS77


![Снимок экрана (94)](https://github.com/microsoft/WSL/assets/123351247/42116586-cc2b-429b-8a53-6ac670d691e1)

**Comment by kotenok2000** [comment]
Try to enable control flow guard for C:\Windows\System32\vmcompute.dll
![image](https://github.com/microsoft/WSL/assets/8822292/c02c3dc8-b011-4422-a7b1-de9e8b9d1849)

**Comment by NuRS7** [comment]
К сожалению не работает...
Bro not working i try

вт, 29 авг. 2023 г., 00:57 Кирилл Фролов ***@***.***>:

**Comment by djehutyy** [comment]
> I dont know... I do not know Kali does not want to work on the first and on the second. Plz help me. TG @t.me/NuRS77
> 
> ![Снимок экрана (94)](https://user-images.githubusercontent.com/123351247/263801889-42116586-cc2b-429b-8a53-6ac670d691e1.png)

Remember to open powershell as administrator
And use wsl 2 instead of 1 
https://learn.microsoft.com/en-us/windows/wsl/install-manual

**Comment by rohithbojja** [comment]
> Installing, this may take a few minutes... WslRegisterDistribution failed with error: 0x80004005 Error: 0x80004005 ??????
> 
> Press any key to continue...
> 
> so is me

run in administator

**Comment by 181726943** [comment]
> 



Does this require me to create a new .wslconfig file?

**Comment by 181726943** [comment]
> AMD processor; the following allowed WSL2 with AMD processor. Seems nestedVirtualization is broken with AMD processors ([#5440 (comment)](https://github.com/microsoft/WSL/issues/5440#issuecomment-778660156))
> 
> %userprofile%.wslconfig
> 
> ```
> [wsl2]
> nestedVirtualization=false
> ```


Does this require me to create a new .wslconfig file?

**Comment by NjiruClinton** [comment]
> 

had same error and this `wsl --update` solved it. figured it might have been something wrong registering the distribution with WSL. seems like the problem might indicate a registration problem conflicting.

**Comment by Eelectro** [comment]
> ` wsl --update` as administrator worked for me
> 
> Then just `wsl --install -d "the distro you want to install"`

thank you! This only way that worked for me.

**Comment by Ajeets6** [comment]
I also had the same error while installing kali and I tried everything but nothing worked, Then I installed ubuntu and that gave me no errors.

**Comment by vidzshan** [comment]
> ` wsl --update` as administrator worked for me

This work well. Thank you bro

**Comment by vidzshan** [comment]
> > ` wsl --update` as administrator worked for me
> 
> This works well. Thank you

**Comment by 0xbinhelal** [comment]
`wsl --update` and then `wsl --install -d kali-linux` worked for me. Don't forget to run as administrator.

**Comment by TawhidsShadow** [comment]
> `wsl --update` and then `wsl --install -d kali-linux` worked for me. Don't forget to run as administrator.

yes this helped me too

**Comment by Exmanq** [comment]
wsl --set-version Ubuntu-22.04 2
wsl --set-version Ubuntu 2

wsl --install -d kali-linux

sudo apt update && sudo apt upgrade -y

sudo apt install -y kali-linux-default
